### PR TITLE
T22836: Import documentation from Google Docs

### PR DESCRIPTION
**Pull request by smspillaz**
https://phabricator.endlessm.com/T22836

### CODE REVIEW

**Comment by smspillaz** [comment]
Documentation reflowed

File: docs/OSIntegration.md [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+In order to avoid this outcome in other areas throughout the OS, we have a set
+of "eks-search-provider" binaries for each SDK compatibility range (for
+instance, v1 is compatible with SDK1, v2 is compatible with SDK2 and SDK3, v3
+is compatible with SDK4).
+
+In reality, apps now talk to the same daemon which dynamically "becomes" the
+correct eks-search-provider binary and SDK combination depending on which one
+was requested. (See:
+[EknServicesMultiplexer](/https//github.com/endlessm/eos-knowledge-services/docs/Architecture.md)
```
Comment (smspillaz) [inline docs/OSIntegration.md]: https://

File: docs/Architecture.md [hunk 2]
```diff
@@ -0,0 +1,555 @@
...
+### Content Database Connection
+Since the Service does not link to the Knowledge Framework
+directly, it needs to maintain a connection to EknServices to
+query content databases. This is done through the
+'EknServicesContentDbConnection` in the `eoscompanion.eknservices_bridge`
+module. That class follows the
+[repository pattern](https://deviq.com/repository-pattern/) - it
+provides a few high level methods to access relevant data as
+a collection without exposing the details of maintaining
+D-Bus connections an and deserializing the data.
```
Comment (smspillaz) [inline docs/Architecture.md]: an and

File: docs/Architecture.md [hunk 3]
```diff
@@ -0,0 +1,555 @@
...
+The `respond_if_error_set` helper can be used to simplify
+error handling code by immediately responding with an error
+if the passed in error is non-NULL, returning `True` if
+a response was set, `False` otherwise.
+
+### Middlewares
+Following the design of [Express](https://expressjs.com/en/guide/using-middleware.html)
+middlewares are used to implement shared logic before route
+handlers are even run.  Even though there is no first-class
+support for Middlewares in libsoup, the Companion App Service
```
Comment (smspillaz) [inline docs/Architecture.md]: Middlewares -> middlewares

File: docs/Architecture.md [hunk 4]
```diff
@@ -0,0 +1,555 @@
...
+                    set to `text/html`.
+ - `custom_response`: Send bytes with a specified content-type.
+
+The `respond_if_error_set` helper can be used to simplify
+error handling code by immediately responding with an error
+if the passed in error is non-NULL, returning `True` if
+a response was set, `False` otherwise.
+
+### Middlewares
+Following the design of [Express](https://expressjs.com/en/guide/using-middleware.html)
```
Comment (smspillaz) [inline docs/Architecture.md]: Express,

File: docs/Architecture.md [hunk 5]
```diff
@@ -0,0 +1,555 @@
...
+to actually transform the read bytes into another byte sequence asynchronously,
+suitable for consumption by the Companion App.
+
+In general, adjusting content is an expensive operation, since it precludes
+the ability to stream content directly from disk using Gio (see
+[Content Streaming and Partial Content](#content-streaming-and-partial-content)
+below). The entire byte sequence also needs to be loaded into memory all at
+once as well.
+
+To use a content adjuster, the caller should create the appropriate
```
Comment (smspillaz) [inline docs/Architecture.md]: Content Adjuster

File: docs/Architecture.md [hunk 6]
```diff
@@ -0,0 +1,555 @@
...
+Since large seekable files can be served by the Service to the Companion App,
+it is important that Service supports the relevant HTTP streaming
+protocols, such that the App can quickly seek through the content
+and only downloads chunks that it requires.
+
+This is achieved through the use of the "Content-Range" and "Partial-Content"
+[specification](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/206).
+The linked document has more information, but in our implementation
+we need to get an uncompressed stream to the file, work out how long that
+stream is (ideally without loading it all into memory), set the
```
Comment (smspillaz) [inline docs/Architecture.md]: Remove "set the"

**Comment by cosimoc** [comment]
Not going to review this line-by-line at this point. Let's merge it. Thanks!

**Pull request merged**